### PR TITLE
misc(url) enforce trailing slash, make sure we are building trailing …

### LINKED ISSUE #3063: Enforce trailing slash in internal linking
After looking at GA i've found an issue with our page views, server side links were different to many front-end links via react router navigations.

Screen Shot 2019-05-24 at 3 37 26 PM

Sometimes on dev we didnt have proper item open in sidebar or anchor links were'nt working.

### PR DESCRIPTION
**Pull request by EugeneHlushko**
Fixes #3063

### CODE REVIEW

File: src/components/Site/Site.jsx [hunk 1]
```diff
@@ -74,6 +74,7 @@ class Site extends React.Component {
           toggle={this._toggleSidebar} /> : null}
 
         <Switch>
+          <Route exact strict path="/:url*" render={props => <Redirect to={`${props.location.pathname}/`}/>} />
```
Comment (montogeek) [inline src/components/Site/Site.jsx]: Why is this needed?

Comment (EugeneHlushko) [reply src/components/Site/Site.jsx]: The context is in the issue, we have two types of Urls, and ssr urls are different from react router urls. When users copy links and share them they share react router links without trailing slashes, which gets into redirect and sometimes anchors didnt work after that e.g. `/concepts#something` ended up on `/concepts/#something`. Also reporting in GA is broken, all page views are duplicated with `/concepts` (landed via react-router) and `/concepts/` (landed from search)

**Comment by montogeek** [comment]
Thanks!
